### PR TITLE
591 : Add support for pagination

### PR DESCRIPTION
**Pull request by rehammuzzamil**
https://github.com/OpenSRP/opensrp-server-web/issues/591

### CODE REVIEW

File: src/main/java/org/opensrp/web/rest/PractitionerRoleResource.java [hunk 1]
```diff
@@ -143,4 +154,18 @@ public void setPractitionerRoleService(PractitionerRoleService practitionerRoleS
...
+        BaseSearchBean.FieldName fieldName;
+        orderByTypeEnum = orderByType != null ? BaseSearchBean.OrderByType.valueOf(orderByType) : BaseSearchBean.OrderByType.DESC;
+        fieldName = orderByFieldName != null ? BaseSearchBean.FieldName.valueOf(orderByFieldName) : BaseSearchBean.FieldName.id;
+
+        return PractitionerRoleSearchBean.builder()
+                .pageNumber(pageNumber)
+                .pageSize(pageSize)
+                .orderByType(orderByTypeEnum)
+                .orderByFieldName(fieldName).build();
+    }
```
Comment (githengi) [inline src/main/java/org/opensrp/web/rest/PractitionerRoleResource.java:169]: Can be refactored to a utils method

File: src/main/java/org/opensrp/web/rest/PractitionerResource.java [hunk 2]
```diff
@@ -102,4 +114,20 @@ public void setPractitionerService(PractitionerService practitionerService) {
...
+        orderByTypeEnum = orderByType != null ? BaseSearchBean.OrderByType.valueOf(orderByType) : BaseSearchBean.OrderByType.DESC;
+        fieldName = orderByFieldName != null ? BaseSearchBean.FieldName.valueOf(orderByFieldName) : BaseSearchBean.FieldName.id;
+
+        return PractitionerSearchBean.builder()
+                .pageNumber(pageNumber)
+                .pageSize(pageSize)
+                .orderByType(orderByTypeEnum)
+                .orderByFieldName(fieldName).build();
+
+    }
```
Comment (githengi) [inline src/main/java/org/opensrp/web/rest/PractitionerResource.java:131]: Can be refactored to a utils method

**Review by githengi — APPROVED** [review]

File: src/main/java/org/opensrp/web/rest/PractitionerResource.java (hunk 2 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/rest/PractitionerResource.java:131]: As per our discussion, using the builder pattern, each populating the builder has to be done for each resource.

File: src/main/java/org/opensrp/web/rest/PractitionerRoleResource.java (hunk 1 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/rest/PractitionerRoleResource.java:169]: As per our discussion, using the builder pattern, each populating the builder has to be done for each resource.

**Pull request merged**